### PR TITLE
Add steps for AWS custom security groups

### PR DESCRIPTION
**Pull request by barbacbd**
** Added the scripts for the security groups to be created and the data to be patched in to the install config.

### CODE REVIEW

**Comment by barbacbd** [comment]
The PR is a reference to https://github.com/openshift/installer/pull/7151

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

File: ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-commands.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+security_group=$(aws ec2 create-security-group --region ${REGION} --description 'CI custom security groups' --group-name "${metadata_name}-sg"  --vpc-id ${vpc_id} --output text)
+echo "${security_group}" > "${sg_file}"
+echo "Security Group: ${security_group}"
+
+CONFIG_PATCH="${SHARED_DIR}/install-config-security-groups.yaml.patch"
+cat > "${CONFIG_PATCH}" << EOF
+compute:
+- platform:
+    aws:
+      additionalSecurityGroupIDs: [$(cat "${sg_file}")]
```
Comment (r4f4) [inline ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-commands.sh:36]: Nit: you already have `${security_group}` so technically you don't need to `cat` from the file where you saved it.

File: ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-commands.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+controlPlane:
+  platform:
+    aws:
+      additionalSecurityGroupIDs: [$(cat "${sg_file}")]
```
Comment (r4f4) [inline ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-commands.sh:40]: Ditto.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by patrickdillon** [comment]
This is only adding steps. We also need to add a job and it would be a good idea to add it in this PR so that rehearsals will be triggered to help test these newly added steps.

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-custom-security-groups

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-custom-security-groups

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-custom-security-groups

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-custom-security-groups

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-custom-security-groups

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Pull request merged**